components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java

```diff
@@ -6463,15 +6463,15 @@ public static String getMountedPath(RegistryContext registryContext, String path
      * @return a Map of domain names for tenant
      * @throws org.wso2.carbon.apimgt.api.APIManagementException if an error occurs when loading tiers from the registry
      */
-    public static Map<String, String> getDomainMappings(String tenantDomain, String appType)
+    public static Map<String, String> getDomainMappings(String organization, String appType)
             throws APIManagementException {
 
         Map<String, String> domains = new HashMap<String, String>();
         String resourcePath;
         try {
             Registry registry = ServiceReferenceHolder.getInstance().getRegistryService().
                     getGovernanceSystemRegistry();
-            resourcePath = APIConstants.API_DOMAIN_MAPPINGS.replace(APIConstants.API_DOMAIN_MAPPING_TENANT_ID_IDENTIFIER, tenantDomain);
+            resourcePath = APIConstants.API_DOMAIN_MAPPINGS.replace(APIConstants.API_DOMAIN_MAPPING_TENANT_ID_IDENTIFIER, organization);
```

Comment:
This won't work in choreo setup.  add this before passing the domain

    if (StringUtils.isNotEmpty(organization)) {
        int tenantId = APIUtil.getInternalOrganizationId(organization);
        // To handle choreo scenario. due to keymanagers are not per organization atm. using ST
        if (tenantId == MultitenantConstants.SUPER_TENANT_ID) {
            organization = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
        }
    }

Reply:
fixed in https://github.com/wso2/carbon-apimgt/pull/10553/commits/92236ff1b77766202a3d3cb139bfe7943a1b81aa